File: security-adapter-in/src/main/java/com/example/security/adapter/in/rest/dto/TenantOnboardRequest.java
package com.example.security.adapter.in.rest.dto;

import com.example.security.domain.tenant.Tenant.PiiMaskingPolicy;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.time.Duration;

public record TenantOnboardRequest(
    @NotBlank String tenantId,
    @NotBlank String displayName,
    @NotNull Duration retention,
    @NotNull Duration hotRetention,
    @NotNull PiiMaskingPolicy piiPolicy) {}

File: security-adapter-in/src/main/kotlin/com/example/security/adapter/in/rest/dto/TenantOnboardRequest.kt
package com.example.security.adapter.`in`.rest.dto

import com.example.security.domain.tenant.Tenant.PiiMaskingPolicy
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import java.time.Duration

/** POST /api/v1/tenants — 새 tenant 등록. */
@JvmRecord
data class TenantOnboardRequest(
    @field:NotBlank val tenantId: String,
    @field:NotBlank val displayName: String,
    @field:NotNull val retention: Duration,
    @field:NotNull val hotRetention: Duration,
    @field:NotNull val piiPolicy: PiiMaskingPolicy,
)
